core/templates/pages/exploration-player-page/layout-directives/exploration-footer.component.spec.ts

```diff
@@ -141,6 +167,315 @@ describe('ExplorationFooterComponent', () => {
     expect(component.hintsAndSolutionsAreSupported).toBeFalse();
   });
 
+  it('should push 1 in checkpointArray for the first card', () => {
+    let ngbModal = TestBed.inject(NgbModal);
+
+    expect(component.checkpointArray).toEqual([0]);
+
+    spyOn(playerPositionService, 'getDisplayedCardIndex').and.
+      returnValue(0);
+
+    spyOn(ngbModal, 'open').and.returnValue({
+      componentInstance: {
+        numberofCheckpoints: 0,
+        completedWidth: 0,
+        contributorNames: [],
+        expInfo: null
+      },
+      result: {
+        then: (successCallback: () => void, errorCallback: () => void) => {
+          successCallback();
+          errorCallback();
+        }
+      }
+    } as NgbModalRef);
+
+    component.openInformationCardModal();
+
+    expect(ngbModal.open).toHaveBeenCalled();
+    expect(component.checkpointArray).toEqual([0, 1]);
+  });
+
+  it('should push 0 in checkpointArray when card_is_checkpoint' +
```

Comment:
Ditto: Prefer to write descriptions so that it reflects the users perspective avoiding implementation details (e.g. variable names). See  https://github.com/oppia/oppia/wiki/Frontend-tests#name-tests-clearly

Reply:
Done!